### ISSUE TITLE
When formatting text from the WYSIWYG editor (not code), synchronising fails to carry over formatting to other clients (which then kills the formatting upon sync).

### ISSUE THREAD

**Issue by overseeing** [laurent22/joplin#3910]
## Environment

Joplin version: 1.2.6 (in macOS and Linux)
Platform: macOS and Linux
OS specifics: Catalina 10.15.7 (19H2) and Linux Mint 20
Joplin Encryption: Yes

## Steps to reproduce

1. Create a new note from the Linux/macOS WYSIWYG editor.
2. Apply formatting in WYSIWYG mode like bold on a word.
3. Synchronise to a second client such as iOS or from Mac to Linux / Linux to Mac.
4. Edit the note (add extra text) in second client, then synchronise back to original client.

## Describe what you expected to happen

3. Should have the second client receiving the rich text formatting. It doesn't.
4. Editing in second client and then syncing back results in all formatting is lost in the original client.

WYSIWYG editor obviously not ready for prime time, but with bug reports like hopefully it can be rock solid! I'll use Joplin (on desktop) in code edit mode for now. Thanks for the work!!

**Comment by laurent22** [comment]
Isn't it a duplicate of #3908 ?

**Comment by mindfulsource** [comment]
Similar, but different. (#3908 I'm still trying to pinpoint.)

This one seems straightforward. If you follow the steps in this OP, this one should be reproducible. I just tried once more on mac and linux client, it happens.

Interestingly, if I *paste* rich formatted text (instead of create it from scratch in the WYSIWYG editor), it retains the formatting in second synced clients (I test linux and iOS as second clients)! (It's even keeping rich text links, indented bullet points, etc.)

This bug is only if you are creating the rich formatting in the mac or linux client in the WYSIWYG editor, e.g. via `cmd-b` or `ctrl-b` to make a word bold.

**Comment by mindfulsource** [comment]
Updated observations:

- If you merely wait some more - after the first sync to a second client - the formatting carries over. If I edit in another client before the formatting successfully syncs over, it kills the formatting permanently in that note. Only once the formatting has carried over, will it not be lost.

- So this is a weird delay. But only sometimes? I just created another note from mac 1.2.6 client in GUI editor, but this time, formatting synced immediately over to iOS and Linux client!

Sigh.

(I've added in OP that I use the Joplin encryption.)
